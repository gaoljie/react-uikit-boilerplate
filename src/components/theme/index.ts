import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
