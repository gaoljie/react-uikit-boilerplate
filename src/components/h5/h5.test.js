import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";
import theme from "../theme";
import H5 from "./index";

describe("Test H5", () => {
  it("H5 Test", () => {
    const wrapper = mount(<H5>Primary</H5>);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper).toHaveStyleRule("color", "#333333");

    expect(wrapper).toHaveStyleRule("font-family", theme.secondaryFont);

    expect(wrapper.children().text()).toEqual("Primary");

    expect(wrapper.exists("h5")).toEqual(true);
  });
});
