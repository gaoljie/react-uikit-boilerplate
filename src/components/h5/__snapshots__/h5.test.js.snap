// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test H5 H5 Test 1`] = `
.c0 {
  font-family: Work Sans;
  color: #333333;
  font-size: 13px;
  line-height: 20px;
}

@media (max-width:769px) {
  .c0 {
    font-size: 12px;
    line-height: 18px;
  }
}

<H5>
  <Styled(styled.div)
    as="h5"
  >
    <StyledComponent
      as="h5"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
              "
  font-family: ",
              "Work Sans",
              ";
  color: ",
              [Function],
              ";
  font-size: 13px;
  line-height: 20px;

  ",
              "@media (max-width: 769px)",
              " {
    font-size: 12px;
    line-height: 18px;
  }
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h5
        className="c0"
      >
        Primary
      </h5>
    </StyledComponent>
  </Styled(styled.div)>
</H5>
`;
